### ISSUE TITLE
Check the data tables used in data bindings exist

### ISSUE THREAD

**Issue by annapowellsmith** [kiln/flourish-sdk#25]
If you reference a non-existing data table in the data binding definition [EDIT: ...and try to publish the template], you get: 

```
😱  Error from server
😱  ERROR:  null value in column "template_data_table_id" violates not-null constraint
DETAIL:  Failing row contains (376, 5, values, filter1, null, 0, null).
```

We should check that the data tables referred to in data bindings really exist, and fail gracefully if not.

**Comment by robinhouston** [comment]
This is the error you get when you publish the template. Presumably something different (a different error?) happens if you do `flourish run`?

But yes, we should validate this.

**Comment by annapowellsmith** [comment]
Yes, `flourish run` gives `Data table id undefined not in [...]`.

**Comment by micycle** [comment]
@annapowellsmith We do already validate data table names, throwing this error:
`template.yml: data binding refers to “Bar::X”, but file does not exist: data/Bar.csv`
Which suggests I don't understand the exact problem. Can I get a copy of the yaml that produced your errors above?

**Comment by robinhouston** [comment]
I wonder what happens if we use a case-insensitive (but case-preserving) filesystem, and there is a case mismatch? I could imagine that causing a problem like this.

**Comment by robinhouston** [comment]
Confirmed!

**Comment by robinhouston** [comment]
This will be fixed in 2.9.0.
